src/topic_tree_.toit

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the LICENSE file.
+
+class TopicTreeNode_:
+  topic_level /string
+  value_ /any := null
+  children /Map ::= {:}  // string -> TopicTreeNode_?
+
+  constructor .topic_level:
+
+/**
+A tree of topics-segments, matching a topic to a value.
+*/
+class TopicTree:
+  root_ /TopicTreeNode_ := TopicTreeNode_ "ignored_root"
+
+  is_empty -> bool:
+    return root_.children.is_empty
+
+  /**
+  Inserts, or replaces the value for the given topic.
+
+  Returns the old value. Null if there was none.
+  */
+  set topic/string value/any -> any:
+    if topic == "": throw "INVALID_ARGUMENT"
```

Comment:
Check that value isn't null? Would using `value/Object` be a bad idea?

Reply:
changed it to `Object`, but that doesn't prevent `null` (as `null` is currently an Object).
Maybe we want to changed that...